File: src/main/java/com/uiptv/model/PublishedM3uCategorySelection.java
package com.uiptv.model;

import com.uiptv.shared.BaseJson;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class PublishedM3uCategorySelection extends BaseJson {
    private String dbId;
    private String accountId;
    private String categoryName;
    private boolean selected;

    public PublishedM3uCategorySelection(String accountId, String categoryName, boolean selected) {
        this.accountId = accountId;
        this.categoryName = categoryName;
        this.selected = selected;
    }
}

File: server/src/main/kotlin/com/uiptv/model/PublishedM3uCategorySelection.kt
package com.uiptv.model

import com.uiptv.shared.BaseJson

data class PublishedM3uCategorySelection @JvmOverloads constructor(
    var dbId: String? = null,
    var accountId: String? = null,
    var categoryName: String? = null,
    @get:JvmName("isSelected")
    var selected: Boolean = false
) : BaseJson() {
    constructor(accountId: String?, categoryName: String?, selected: Boolean) : this(null, accountId, categoryName, selected)
}
